utils/utilities.js

```diff
@@ -138,12 +175,102 @@ module.exports = {
 
   personalSign: async (signHash, signer) => ethers.utils.joinSignature(signer.signingKey.signDigest(signHash)),
 
-  callStatic: async (contractWrapper, method, ...args) => {
-    const contract = new ethers.Contract(
-      contractWrapper.contractAddress,
-      contractWrapper.contract.interface.abi,
-      ethers.getDefaultProvider(contractWrapper.provider.connection.url),
-    );
-    return contract.callStatic[method](...args);
+  async getBalance(account) {
+    const balance = await web3.eth.getBalance(account);
+    return new BN(balance);
+  },
+
+  async getTimestamp(blockNumber) {
+    const blockN = !blockNumber ? "latest" : blockNumber;
+    const { timestamp } = await web3.eth.getBlock(blockN);
+    return timestamp;
+  },
+
+  async getChainId() {
+    // TODO: The web3 version packaged with truffle is 1.2.1 while the getChainId logic
+    // we need here was introduced in 1.2.2
+    // Uncomment when https://github.com/trufflesuite/truffle/issues/2688#issuecomment-709879003 is resolved
+    // const chainId = await web3.eth.getChainId();
+    // console.log("chainId", chainId)
+    // return chainId;
+    return 1895;
   },
+
+  async web3GetClient() {
+    return new Promise((resolve, reject) => {
+      web3.eth.getNodeInfo((err, res) => {
+        if (err !== null) return reject(err);
+        return resolve(res);
+      });
+    });
+  },
+
+  async increaseTime(seconds) {
+    const client = await this.web3GetClient();
+    const p = new Promise((resolve, reject) => {
+      if (client.indexOf("TestRPC") === -1) {
+        console.warning("Client is not ganache-cli and cannot forward time");
+      } else {
+        web3.currentProvider.send(
+          {
+            jsonrpc: "2.0",
+            method: "evm_increaseTime",
+            params: [seconds],
+            id: 0,
+          },
+          (err) => {
+            if (err) {
+              return reject(err);
+            }
+            return web3.currentProvider.send(
+              {
+                jsonrpc: "2.0",
+                method: "evm_mine",
+                params: [],
+                id: 0,
+              },
+              (err2, res) => {
+                if (err2) {
+                  return reject(err2);
+                }
+                return resolve(res);
+              }
+            );
+          }
+        );
+      }
+    });
+    return p;
+  },
+
+  async getNonceForRelay() {
+    const block = await web3.eth.getBlockNumber();
+    const timestamp = new Date().getTime();
+    return `0x${ethers.utils.hexZeroPad(ethers.utils.hexlify(block), 16)
+      .slice(2)}${ethers.utils.hexZeroPad(ethers.utils.hexlify(timestamp), 16).slice(2)}`;
+  },
+
+  async assertRevert(promise, revertMessage) {
```

Comment:
Is there a benefit in using our own function here instead of the one from `truffle-assertions` ?(https://github.com/rkalis/truffle-assertions/blob/master/index.js#L146)

Reply:
I hadn't heard of `truffle-assertions`. Works well to replace the utils method. Updated.

Reply:
Updated the `assertRevert` usage with `truffle-assertions.reverts`. I experimented with `eventEmitted` as well but that still isn't able to decode nested events (from our RelayerManager) but this missing feature seems to be tracked here https://github.com/rkalis/truffle-assertions/issues/6